pkg/ebpf/generic_map_test.go

```diff
@@ -0,0 +1,444 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+
+package ebpf
+
+import (
+	"fmt"
+	"testing"
+
+	ebpfkernel "github.com/DataDog/datadog-agent/pkg/security/ebpf/kernel"
+	"github.com/DataDog/datadog-agent/pkg/util/kernel"
+	"github.com/cilium/ebpf"
+	"github.com/stretchr/testify/require"
+)
+
+func TestBatchAPISupported(t *testing.T) {
+	// Batch API is supported on kernels >= 5.6, so make sure that in those cases
+	// it returns true
+	kernelVersion, err := ebpfkernel.NewKernelVersion()
+	require.NoError(t, err)
+
+	if kernelVersion.IsRH7Kernel() || kernelVersion.IsRH8Kernel() {
+		// Some of those kernels have backported the batch API, I don't want
+		// to include those specifics in unit tests that are only meant to ensure
+		// that the checks are correct in at least the basic cases.
+		t.Skip("Unknown support for batch API on RHEL kernels")
+	}
+
+	if kernelVersion.Code < ebpfkernel.Kernel5_9 {
```

Comment:
The comment on line 21 says the support is from 5.6 and above
```suggestion
	if kernelVersion.Code < ebpfkernel.Kernel5_6 {
```